lib/resources/odata.js

```diff
@@ -46,13 +46,13 @@ module.exports = (service, endpoint) => {
       return matches[1];
     };
 
-    const singleRecord = endpoint.odata.json(({ Forms, Submissions, env }, { auth, params, query, originalUrl }) =>
-      getForm(Forms, auth, params)
-        .then((form) => Promise.all([
-          Forms.getFields(form.def.id).then(selectFields(query, getTableFromOriginalUrl(originalUrl))),
-          Submissions.getForExport(form.id, getUuid(params.uuid), draft).then(getOrNotFound), // may require s3 blob handling
-        ])
-          .then(([fields, row]) => singleRowToOData(fields, row, env.domain, originalUrl, query))));
+    const singleRecord = endpoint.odata.json(async ({ Forms, Submissions, env }, { auth, params, query, originalUrl }) => {
+      const form = await getForm(Forms, auth, params);
+      const allFields = await Forms.getFields(form.def.id);
+      const selectedFields = await selectFields(query, getTableFromOriginalUrl(originalUrl))(allFields);
```

Comment:
I think we can remove `await` from here since it is synchronous/non-promise function 

```suggestion
      const selectedFields = selectFields(query, getTableFromOriginalUrl(originalUrl))(allFields);
```